server/boot/user.js

```diff
@@ -266,6 +273,53 @@ module.exports = function(app) {
     return res.redirect('/' + username);
   }
 
+  function getUnlinkSocial(req, res, next) {
+    const { user } = req;
+    const { username } = user;
+    let social = req.params.social;
+    if (!social) {
+      req.flash('errors', {
+        msg: 'No social account found'
+      });
+      return res.redirect('/' + username);
+    }
+
+    social = social.toLowerCase();
+    const validSocialAccounts = ['twitter', 'linkedin'];
+    if (validSocialAccounts.indexOf(social) === -1) {
+      req.flash('errors', {
+        msg: 'Invalid social account'
+      });
+      return res.redirect('/' + username);
+    }
+
+    if (!user[social]) {
+      req.flash('errors', {
+        msg: `No ${social} account associated`
+      });
+      return res.redirect('/' + username);
+    }
+
+    const query = {};
+    query['userId'] = user.id;
+    query['provider'] = social;
+    return UserIdentity.destroyAll(query, function(err) {
```

Comment:
I don't feel safe using an operator that could potential destroy all the data.

Let's use the relationship operator on the user object.

I believe it is
```js
user.userIdentities({ provider: social }).then(....)
```

Then 
```js
user.userIdentities.destroyById(userIdentId, ...
```

This will make sure nothing could be accidentally destroyed.